### PR TITLE
Set up Bolt SSO buttons on Magento frontend

### PR DESCRIPTION
**Pull request by nickfiacco-bolt**
For Bolt SSO, we replace the native sign in and register buttons with our own.
This PR adds the Bolt SSO button to the header conditionally depending on whether the
feature switch and config value are set to enable the functionality.

# Description
Please include a summary of the change and which issue is fixed. Include the motivation for the changes, and comment on your PR if necessary for clarity.

Fixes: (link Jira ticket)

#changelog Set up Bolt SSO buttons on Magento frontend

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

**Comment by adentong** [comment]
Also update etc/config.xml and getAllConfigSettings?

**Review by adentong — COMMENTED** [review]
LGTM, but I'll let people more knowledgeable about this approve.

File: Block/Js.php
Comment (danac-gs) [inline Block/Js.php:600]: Shoppers are usually addressed as customers in Magento

File: Block/Js.php
Comment (danac-gs) [reply Block/Js.php:604]: Not 100% sure, seems OK, but we should double check / test.

File: etc/events.xml
Comment (danac-gs) [inline etc/events.xml:19]: This observer should be declared in etc/frontend/events.xml so it doesn't unnecessarily execute in admin.

**Review by danac-gs — DISMISSED** [review]
Looks pretty good overall. A few small requests.

**Review by vitaliyreznikov — APPROVED** [review]
LGTM

**Comment by adentong** [comment]
@danac-gs I made some changes according to your comments. Want to get your approval before I merge.

File: Block/Js.php
Comment (mpchadwick) [reply Block/Js.php:517]: Hey - I'm doing a code review of your module for a client right now and just to let you know the cache is sharded based on the HTTP Context. So there are multiple copies of the page cached based on parameters such as whether the user is logged in or what customer group they are in. This allows Magento to implement features such as customer group based pricing.

**Pull request merged**
